File: codewars/7kyu/factorial/dduck-factorial.js
// https://www.codewars.com/kata/factorial-1/train/javascript

const factorial = (n, sum = 1) => (n ? factorial(n - 1, sum * n) : sum);

console.log(factorial(0), 1);
console.log(factorial(1), 1);
console.log(factorial(4), 24);
console.log(factorial(7), 5040);
console.log(factorial(17), 355687428096000);

File: codewars/7kyu/factorial/dduck-factorial.py
# https://www.codewars.com/kata/factorial-1/train/javascript

factorial = lambda n, sum=1: factorial(n - 1, sum * n) if n else sum

print(factorial(0), 1);
print(factorial(1), 1);
print(factorial(4), 24);
print(factorial(7), 5040);
print(factorial(17), 355687428096000);
